src/main/diagnostics/diagnostics.ts

```diff
@@ -54,9 +55,40 @@ export class Diagnostics {
   /**
    * @param categoryName {string}
    * @param id {string}
-   * Returns {unknown} if the categoryName isn't known or id not in that category, the check object otherwise.
+   * Returns an array of all matching checkObjects, depending on which of categoryName and id are specified.
    */
-  getCheckByID(categoryName: string, id: string): DiagnosticsCheck {
-    return this.checksByCategory[categoryName]?.filter(check => check.id === id)[0];
+  getChecks(categoryName: string|null, id: string|null): DiagnosticsCheck[] {
+    if (categoryName) {
+      const checksByCategory = this.checksByCategory[categoryName];
+
+      if (!checksByCategory) {
+        return [];
+      } else if (!id) {
+        return augmentChecks(categoryName, checksByCategory);
```

Comment:
Any reason not to normalize all checks on load (that is, call `agumentChecks` early so that we don't have to do this per call)?

Reply:
Makes sense.